### PR TITLE
refactor(design-system): merge n8n square button into n8n button

### PR DESCRIPTION
**Pull request by cstuncsik**

### CODE REVIEW

File: packages/design-system/src/components/N8nButton/Button.vue [hunk 1]
```diff
@@ -444,4 +454,9 @@ $loading-overlay-background-color: rgba(255, 255, 255, 0);
 .float-right {
 	float: right;
 }
+
+.square {
+  justify-content: center;
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nButton/Button.vue:459]: nitpick - spacing

File: packages/design-system/src/components/N8nButton/Button.vue [hunk 2]
```diff
@@ -75,6 +75,10 @@ export default Vue.extend({
 			validator: (value: string): boolean =>
 				['left', 'right'].includes(value),
 		},
+	  square: {
+		  type: Boolean,
+		  default: false,
+	  },
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nButton/Button.vue:81]: why do we need a square prop? buttons are already now square...

File: packages/design-system/src/components/N8nButton/Button.vue [hunk 3]
```diff
@@ -444,4 +454,9 @@ $loading-overlay-background-color: rgba(255, 255, 255, 0);
 .float-right {
 	float: right;
 }
+
+.square {
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nButton/Button.vue:458]: is this class needed? are not we centering by default?

File: packages/design-system/src/components/N8nButton/Button.vue (hunk 1 above)
Comment (cstuncsik) [reply packages/design-system/src/components/N8nButton/Button.vue]: removed the selector

File: packages/design-system/src/components/N8nButton/Button.vue (hunk 2 above)
Comment (cstuncsik) [reply packages/design-system/src/components/N8nButton/Button.vue:81]: they are square only if the label is exactly one character, in the case of multiple characters if we want to keep the square shape we need a prop to differentiate

File: packages/design-system/src/components/N8nButton/Button.vue (hunk 3 above)
Comment (cstuncsik) [reply packages/design-system/src/components/N8nButton/Button.vue]: actually not, in the case of icon only button the shape is already square but the icon is not centered (icon has the same position as if the button contained text as well)

but replacing `.square` with `&:not(.text)` and putting it under `.icon` is having the same effect but the `square` prop and class name are still necessary for other reasons already mentioned

File: packages/design-system/src/components/N8nButton/Button.vue (hunk 2 above)
Comment (mutdmour) [reply packages/design-system/src/components/N8nButton/Button.vue:81]: I see.. can we delete `icon-button` class from here then and set `square` on IconButton component?

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.194.0`

**Pull request merged**